components/layout/positioned.rs

```diff
@@ -104,55 +94,26 @@ impl AbsolutelyPositionedBox {
     }
 }
 
-impl IndependentFormattingContext {
-    #[inline]
-    pub(crate) fn new_positioning_context(&self) -> Option<PositioningContext> {
-        self.base.new_positioning_context()
-    }
-}
-
-impl LayoutBoxBase {
-    #[inline]
-    pub(crate) fn new_positioning_context(&self) -> Option<PositioningContext> {
-        PositioningContext::new_for_style(&self.style, &self.base_fragment_info.flags)
-    }
+#[derive(Clone, Default, MallocSizeOf)]
+pub(crate) struct PositioningContext {
+    absolutes: Vec<HoistedAbsolutelyPositionedBox>,
 }
```

Comment:
Could we add a doc comment here to document what a "PositioningContext" is/does? i.e. that it is a container that position:absolute and position:fixed nodes are pushed into so that they can be returned up the tree with their static layouts adjusted at each step and their final layouts computed at appropriate boundaries.